### PR TITLE
React-wrapped oncoprint component

### PR DESCRIPTION
**Pull request by adamabeshouse**

### CODE REVIEW

File: src/pages/resultsView/ResultsViewPageStore.ts [hunk 1]
```diff
@@ -798,19 +1206,302 @@ export class ResultsViewPageStore {
...
+            },{} as { [hugoGeneSymbol:string] : Mutation[] });
+        },
+        onResult: (result:{[hugoGeneSymbol: string]: Mutation[]})=>{
+            for (const gene of Object.keys(result)) {
+                this.mutationDataCache.addData([result[gene]]);
+            }
+        }
+    });
+
+    // Mutation annotation
```
Comment (onursumer) [inline src/pages/resultsView/ResultsViewPageStore.ts]: I think at some point we should reorganize our store structure. For example, we can have a `MutationAnnotationStore` for this kind of shared data. We can then pass it to both mutation mapper store and oncoprint store.

Comment (adamabeshouse) [reply src/pages/resultsView/ResultsViewPageStore.ts]: im open to that

File: src/shared/components/oncoprint/ResultsViewOncoprint.tsx [hunk 2]
```diff
@@ -0,0 +1,808 @@
...
+                !this.props.store.mutationAnnotationSettings.cbioportalCount &&
+                !this.props.store.mutationAnnotationSettings.cosmicCount) {
+                this.distinguishDrivers = false;
+                this.props.store.mutationAnnotationSettings.ignoreUnknown = false;
+            } else {
+                this.distinguishDrivers = true;
+            }
+        });
+
+        return observable({
```
Comment (alisman) [inline src/shared/components/oncoprint/ResultsViewOncoprint.tsx]: probably doesn't need to be observable unless these functions are going to be changed.

File: src/shared/components/oncoprint/ResultsViewOncoprint.tsx [hunk 3]
```diff
@@ -0,0 +1,808 @@
...
+                this.molecularProfileIdToHeatmapTracks.clear();
+                if (this.sortMode.type === "heatmap") {
+                    this.sortByData();
+                }
+            }),
+            onClickClusterHeatmap:()=>{
+                this.sortMode = {type: "heatmap", clusteredHeatmapProfile: this.selectedHeatmapProfile};
+            },
+            onClickDownload:(type:string)=>{
+                if (type === "pdf") {
```
Comment (alisman) [inline src/shared/components/oncoprint/ResultsViewOncoprint.tsx]: switch statement

File: src/shared/components/oncoprint/SortUtils.ts [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+    for (let i=0; i<array_spec.length; i++) {
+        const equiv_values = ([] as any[]).concat(array_spec[i]);
+        for (let j=0; j<equiv_values.length; j++) {
+            metric[equiv_values[j]] = i;
+        }
+    }
+    return metric;
+};
+
+function sign(x:number):0|-1|1 {
```
Comment (alisman) [inline src/shared/components/oncoprint/SortUtils.ts]: duplicate?

File: src/shared/components/oncoprint/ResultsViewOncoprint.tsx (hunk 2 above)
Comment (adamabeshouse) [reply src/shared/components/oncoprint/ResultsViewOncoprint.tsx]: note to self: doen

File: src/shared/components/oncoprint/ResultsViewOncoprint.tsx (hunk 3 above)
Comment (adamabeshouse) [reply src/shared/components/oncoprint/ResultsViewOncoprint.tsx]: note to self: done

File: src/shared/components/oncoprint/SortUtils.ts (hunk 4 above)
Comment (adamabeshouse) [reply src/shared/components/oncoprint/SortUtils.ts]: strangely enough a grep for "function sign" and "function sgn" dont turn up anything

**Pull request merged**